utils/wasm-builder/src/lib.rs

```diff
@@ -159,6 +164,25 @@ impl WasmBuilder {
         }
         Ok(matched_features)
     }
+
+    fn paths_to_remap(&self) -> Result<Vec<(PathBuf, &'static str)>> {
```

Comment:
I'd do this remap for production builds only because with debug features etc it's useful to have ability to tap on real path (e.g. when panic occurred) and your idea goes to the file

Reply:
Maybe builder flag?

Reply:
We have `self.wasm_project.profile()`. So, maybe if the smart contract is compiled with the "dev"/"debug" profile, it will have full paths. When you build contract with "release" profile, it will remap the paths.

Reply:
> We have `self.wasm_project.profile()`. So, maybe if the smart contract is compiled with the "dev"/"debug" profile, it will have full paths. When you build contract with "release" profile, it will remap the paths.

That is exactly I'm talking about

Reply:
@breathx should be resolved now